packages/block-library/src/image/test/edit.native.js

```diff
@@ -196,25 +195,16 @@ describe( 'Image Block', () => {
 
 		const [ imageBlock ] = screen.getAllByLabelText( /Image Block/ );
 		fireEvent.press( imageBlock );
-		// Awaiting navigation event seemingly required due to React Navigation bug
-		// https://github.com/react-navigation/react-navigation/issues/9701
-		await act( () =>
-			fireEvent.press( screen.getByLabelText( 'Open Settings' ) )
-		);
+		fireEvent.press( screen.getByLabelText( 'Open Settings' ) );
+
 		fireEvent.press( screen.getByText( 'None' ) );
-		fireEvent.press( screen.getByText( 'Media File' ) );
```

Comment:
IIRC, this test's original intention was to verify the following flow: `None` → `Media File` → `Custom URL` → `Media File`. This was done because a scenario where not _all_ properties were correctly updated during the flow. 

Because of that, it may make most sense to reinstate `Media File` via two presses on that text. Once to select `Media File`, another to navigate back to the list of options. WDYT? 

<details><summary>Video of Flow</summary>

https://github.com/WordPress/gutenberg/assets/438664/bf8bd27f-1659-49ad-81cc-19c717780584

</details>

Reply:
I'll have a second look at this test in the next few days.

I merged this PR now because I was afraid that it would get stalled again for a long time if I wanted to make everything perfect before merging.